### ISSUE TITLE
A "catch-all" route causes a panic

### ISSUE THREAD

**Issue by markbates** [gobuffalo/buffalo#561]
```go
app.GET("/{name:.+}", func(c buffalo.Context) error {
	fmt.Println(c.Param("name"))
	return c.Redirect(302, "/")
})
```

```
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
github.com/markbates/inflect.(*Ruleset).CamelizeDownFirst(0xc420126e70, 0xc4201cf75b, 0x0, 0x0, 0x0)
	/Users/markbates/Dropbox/development/gocode/src/github.com/markbates/inflect/inflect.go:388 +0xd3
github.com/markbates/inflect.CamelizeDownFirst(0x0, 0x0, 0x0, 0x0)
	/Users/markbates/Dropbox/development/gocode/src/github.com/markbates/inflect/inflect.go:602 +0x41
github.com/gobuffalo/buffalo.buildRouteName(0x47d4e1f, 0xa, 0x1, 0x1)
	/Users/markbates/Dropbox/development/gocode/src/github.com/gobuffalo/buffalo/router.go:219 +0x67b
github.com/gobuffalo/buffalo.(*App).addRoute(0xc4201477a0, 0x47ce1cd, 0x3, 0x47d4e1f, 0xa, 0x47f7da0, 0x0)
	/Users/markbates/Dropbox/development/gocode/src/github.com/gobuffalo/buffalo/router.go:170 +0x3fe
github.com/gobuffalo/buffalo.(*App).GET(0xc4201477a0, 0x47d4e1f, 0xa, 0x47f7da0, 0x0)
	/Users/markbates/Dropbox/development/gocode/src/github.com/gobuffalo/buffalo/router.go:16 +0x5e
github.com/markbates/coke/actions.App(0x47ceaba)
	/Users/markbates/Dropbox/development/gocode/src/github.com/markbates/coke/actions/app.go:62 +0x38f
main.main()
	/Users/markbates/Dropbox/development/gocode/src/github.com/markbates/coke/main.go:12 +0x68
```

**Comment by apaganobeleno** [comment]
@markbates how do you think should be the name of that route ?

**Comment by markbates** [comment]
I would call it "unnamed". If a second one comes in "unnamed2", etc…

**Comment by apaganobeleno** [comment]
ok, i will just name the route "unnamed" as we discussed on Slack, if there is another route with the same shape it would override this one, so it should have the same name.

**Comment by apaganobeleno** [comment]
closed by #562
